.github/workflows/pkgci_test_android.yml

```diff
@@ -0,0 +1,154 @@
+      - uses: actions/download-artifact@v4.1.7
+        with:
+          name: linux_x86_64_release_packages
+          path: ${{ env.PACKAGE_DOWNLOAD_DIR }}
+      - name: Setup base venv
+        run: |
+          ./build_tools/pkgci/setup_venv.py ${VENV_DIR} \
+            --artifact-path=${PACKAGE_DOWNLOAD_DIR} \
+            --fetch-gh-workflow=${{ inputs.artifact_run_id }}
+      - name: Install build dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y cmake clang ninja-build
+          export CC=clang
+          export CXX=clang++
+
+      # Android-specific setup.
+      - uses: nttld/setup-ndk@v1
+        with:
+          ndk-version: r25b
+
+      # Build then upload.
+      - name: Build for Android
+        env:
+          IREE_HOST_BIN_DIR: ${{ env.VENV_DIR }}/bin
+          IREE_READ_REMOTE_CCACHE: 0
+          IREE_WRITE_REMOTE_CCACHE: 0
+          IREE_READ_LOCAL_CCACHE: 1
+          IREE_WRITE_LOCAL_CCACHE: ${{ inputs.write-caches }}
+          BUILD_PRESET: test
+        run: build_tools/cmake/build_android.sh
+      - name: "Uploading target build dir archive"
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-android-arm_64
+          path: ${{ env.IREE_TARGET_BUILD_DIR }}
+          if-no-files-found: error
+
+  test:
+    # These physical devices are not scalable. Only run on postsubmit for now.
+    # if: (! inputs.is-pr)  # re-enable before merge
+    needs: cross_compile
+    strategy:
+      matrix:
+        target:
+          - device-name: pixel-6-pro
+            label-exclude: "^requires-gpu"
+          - device-name: moto-edge-x30
+            # Moto Edge X30 supports VK_KHR_16bit_storage for only storage
+            # buffers, but not uniform buffers, see
+            # https://vulkan.gpuinfo.org/displayreport.php?id=14481#features_extensions
+            # We request both bits. Disable running such tests.
+            label-exclude: "^requires-gpu|vulkan_uses_vk_khr_16bit_storage"
+    name: test_on_${{ matrix.target.device-name }}
+    runs-on:
+      - self-hosted # must come first
+      - runner-group=${{ inputs.runner-group }}
+      - environment=${{ inputs.runner-env }}
+      - machine-type=${{ matrix.target.device-name }}
+    env:
+      TARGET_BUILD_DIR: build-android-arm_64
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@v4.1.7
+      - uses: actions/download-artifact@v4.1.7
+        with:
+          name: build-android-arm_64
+          path: ${{ env.TARGET_BUILD_DIR }}
+      - name: "Examining artifact directory structure"
+        run: ls -R ${{ env.TARGET_BUILD_DIR }}
+      - name: "Examining github workspace directory structure"
+        run: ls -R ${{ github.workspace }}
+      - name: "Running tests"
+        env:
+          LABEL_EXCLUDE: ${{ matrix.target.label-exclude }}
+        run: |
+          ctest -j 4 \
+            --test-dir "${TARGET_BUILD_DIR}" \
```

Comment:
I'm having some trouble getting into a productive debugging workflow here. 

Logs: https://github.com/iree-org/iree/actions/runs/10188925129/job/28186536935
```
Internal ctest changing into directory: /runner-root/actions-runner/_work/iree/iree/build-android-arm_64
Test project /runner-root/actions-runner/_work/iree/iree/build-android-arm_64
          Start    1: iree/tools/test/iree_run_module_correctness_test_on_android_device
Could not find executable /home/runner/work/iree/iree/build_tools/cmake/run_android_test.sh
Looked in the following places:
Unable to find required file: /home/runner/work/iree/iree/build-android-arm_64/tools/test/iree_run_module_bytecode_module_vmvx.vmfb
```

May be able to test with packages from nightly releases / pypi and a non-cross-compile job doing the same steps:
1. on runner A: build, upload build dir
2. on runner B: download build dir, run ctest

This _should_ be simple, but something about the paths isn't lining up the way it did when using GCS

Reply:
Test code for history / if someone wants to pick up that work: https://github.com/ScottTodd/iree/blob/540b3ed77849a2792e6f33f552c4ab37ba9c01e2/.github/workflows/pkgci_test_android.yml#L94-L154